### DISCUSSION TITLE
How to simplify each tree after generation?

### DISCUSSION THREAD

**Discussion by foxtran** [MilesCranmer/PySR#516] [Q&A]
Sometimes, SymbolicRegression.jl generates something like `unary_op(constant)` and such expressions are not simplified. Is there any way to enforce simplifying after each tree generation?

**Comment by MilesCranmer** [comment]
Sorry, I missed this for some reason. The simplification should happen regularly but it is a separate step. You actually don't want to simplify *every* step because it can prevent certain expressions from being found which are one hop away. e.g., `unary_op(constant)` might become `unary_op(constant + x)`. But if you simplify beforehand it wouldn't be able to necessary get there.
